### PR TITLE
FOGL-9108: Fixed integer and float directive to accept valid input

### PR DESCRIPTION
**Pull request by MonikaSharma06**

### CODE REVIEW

File: src/app/directives/float-only.directive.ts [hunk 1]
```diff
@@ -1,23 +1,72 @@
...
+    }
+  }
+
+  isVaildFloat(event) {
+    const allowedChars = /[0-9.]/;
+    const inputChar = String.fromCharCode(event.charCode);
+    const isAllowedKeys = ['Backspace', 'Tab', 'ArrowLeft', 'ArrowRight', 'Delete', 'e', 'E', '-', '+'].includes(event.key);
+    const isScientificNotation = event.key.toLowerCase() === 'e';
+    const hasAlreadyScientificNotation = this.el.nativeElement.value.includes('e') || this.el.nativeElement.value.includes('E');
+    if (allowedChars.test(inputChar) || isAllowedKeys) {
```
Comment (praveen-garg) [inline src/app/directives/float-only.directive.ts:39]: Here we have regex check for  /[0-9.]/; Cannot we have same for int?
or
event.key here?

**Review by praveen-garg — COMMENTED** [review]
Can we have a directive number-only and have common code there; Override that for int and float?

File: src/app/directives/float-only.directive.ts [hunk 2]
```diff
@@ -1,23 +1,72 @@
...
     if (inputChar === '.' && (event.target as HTMLInputElement).value.includes('.')) {
       event.preventDefault();
     }
+
+    if (event) {
+      return this.isVaildFloat(event);
+    }
+  }
+
+  isVaildFloat(event) {
```
Comment (hemantgarg923) [inline src/app/directives/float-only.directive.ts:33]: `isValidFloat` typo

File: src/app/directives/float-only.directive.ts (hunk 1 above)
Comment (MonikaSharma06) [reply src/app/directives/float-only.directive.ts]: Yes, we can have same also.

> Can we have a directive number-only and have common code there; Override that for int and float?

Yes, we can try in that way also but to keep these separate seems better.

**Review by hemantgarg923 — DISMISSED** [review]

**Comment by praveen-garg** [comment]
> Can we have a directive number-only and have common code there; Override that for int and float?

@mshariq-nerd thoughts?

**Review by hemantgarg923 — APPROVED** [review]

**Comment by mshariq-nerd** [comment]
> > Can we have a directive number-only and have common code there; Override that for int and float?
> 
> @mshariq-nerd thoughts?

Either way is fine. If we will have a number-only directive in that from every configuration item of type int & float user need to pass the type as well to check if float or number.

**Review by mshariq-nerd — APPROVED** [review]

**Pull request merged**